contracts/src/multisig.cairo

```diff
@@ -0,0 +1,501 @@
+use array::ArrayTrait;
+use option::OptionTrait;
+
+fn assert_unique_values<T,
+impl TCopy: Copy::<T>,
+impl TDrop: Drop::<T>,
+impl TPartialEq: PartialEq::<T>,
+>(
+    a: @Array::<T>
+) {
+    let len = a.len();
+    _assert_unique_values_loop(a, len, 0_usize, 1_usize);
+}
+
+fn _assert_unique_values_loop<T,
+impl TCopy: Copy::<T>,
+impl TDrop: Drop::<T>,
+impl TPartialEq: PartialEq::<T>,
+>(
+    a: @Array::<T>, len: usize, j: usize, k: usize
+) {
+    if j >= len {
+        return ();
+    }
+    if k >= len {
+        gas::withdraw_gas_all(get_builtin_costs()).expect('Out of gas');
```

Comment:
We only need one call to withdraw_gas_all in the beginning of the function. (ref: https://medium.com/nethermind-eth/under-the-hood-of-cairo-1-0-exploring-sierra-7f32808421f5). It's meant to calulate the gas cost of the rest of the function and subtracts it from the local gas wallet.

Reply:
Hmm I've always put it at the end of the function (just before the recursive call) since that seemed to be what they do in the examples. We can soon remove this because `alpha.7` will automatically add these